tests/test_run_reference_vmm.py

```diff
@@ -139,6 +144,17 @@ def shutdown(vmm_process):
     vmm_process.wait(timeout=3)
 
 
+def shutdown_piped(vmm_process, vmm_input_fd):
+    os.write(vmm_input_fd, b'reboot -f\n')
+
+    # If the process hasn't ended within 3 seconds, this will raise a
+    # TimeoutExpired exception and fail the test.
+    vmm_process.wait(timeout=3)
+
+    os.remove(NAMED_PIPES["in"])
+    os.remove(NAMED_PIPES["out"]) 
```

Comment:
We should remove them when the test fails as well. I think a better approach would be to not have them as global variables, and just define them in the test. Then, in the test we can initialize them to temporary files. It looks like this can be achieved with https://docs.pytest.org/en/6.2.x/tmpdir.html, but I haven't looked in depth at it.